momepy/coins.py

```diff
@@ -378,118 +378,37 @@ def _list_to_pairs(in_list):
     return [list(pair) for pair in zip(tmp_list[:-1], tmp_list[1:], strict=True)]
 
 
-def _compute_angle(point1, point2):
-    """Calculates the angle between two points in space."""
-    height = abs(point2[1] - point1[1])
-    base = abs(point2[0] - point1[0])
-    angle = round(math.degrees(math.atan(height / base)), 3)
-    return angle
-
-
 def _compute_orientation(line):
-    """Calculates the orientation of a line segment. Point1 is
-    the lower one on the y-axes and vice versa for Point2.
-    """
-    point1 = line[1]
-    point2 = line[0]
-
-    # If the latutide of a point is less and the longitude is more, or
-    # If the latitude of a point is more and the longitude is less, then
-    # the point is oriented leftward and wil have negative orientation.
-    if ((point2[0] > point1[0]) and (point2[1] < point1[1])) or (
-        (point2[0] < point1[0]) and (point2[1] > point1[1])
-    ):
-        return -_compute_angle(point1, point2)
-
-    # if the latitudes are same, the line is horizontal
-    elif point2[1] == point1[1]:
-        return 0
-
-    # if the longitudes are same, the line is vertical
-    elif point2[0] == point1[0]:
-        return 90
+    """Calculates the orientation of a line segment."""
+    point1 = line[0]
+    point2 = line[1]
+    dy = point1[1] - point2[1]
+    dx = point1[0] - point2[0]
+    return math.degrees(math.atan2(dy, dx))
 
-    return _compute_angle(point1, point2)
 
-
-def _points_set_angle(line1, line2):
-    """Calculate the acute joining angle between two given set of points."""
-    l1orien = _compute_orientation(line1)
-    l2orien = _compute_orientation(line2)
-
-    if ((l1orien > 0) and (l2orien < 0)) or ((l1orien < 0) and (l2orien > 0)):
-        return abs(l1orien) + abs(l2orien)
-
-    elif ((l1orien > 0) and (l2orien > 0)) or ((l1orien < 0) and (l2orien < 0)):
-        theta1 = abs(l1orien) + 180 - abs(l2orien)
-        theta2 = abs(l2orien) + 180 - abs(l1orien)
-        if theta1 < theta2:
-            return theta1
-        else:
-            return theta2
-
-    elif (l1orien == 0) or (l2orien == 0):
-        if l1orien < 0:
-            return 180 - abs(l1orien)
-        elif l2orien < 0:
-            return 180 - abs(l2orien)
-        else:
-            return 180 - (
-                abs(_compute_orientation(line1)) + abs(_compute_orientation(line2))
-            )
-
-    elif l1orien == l2orien:
-        return 180
+def _compute_angle(a, b, c):
```

Comment:
I have noticed there was also a function to compute the angle included between three points here:

https://github.com/pysal/momepy/blob/c3c5d5ee704a2fa5d3046e2b4be14c0832d93be9/momepy/utils.py#L103-L110

However, to get the angle included between the points ("a", "b", "c") with "b" being the vertex I had to switch the order of "a" and "b" - or am I overlooking something?

Reply:
(see also my comment above) if we're using the vector logic, we don't need to compute angles between points (again, this is an artifact of having the horizontal plane as reference) and can go directly to computing angles between vectors. and since by definition the 2 vectors start in the same point, we can define their direction as both pointing away from the origin. so @fnattino exactly, agree that the order is relevant, *however* we luckily don't need to consider this if we use vector logic for angle computation